server/pom.xml

```diff
@@ -0,0 +1,417 @@
+                    </configuration>
+                  </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <!-- To know which version of your application you have deployed on
+                    a machine -->
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>3.0.2</version>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
+                        </manifest>
+                    </archive>
+                </configuration>
+            </plugin>
+            <plugin>
+                <!-- Creating a fat jar by including all classes required for running
+                    the app -->
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>3.0.0</version>
+                <configuration>
+                    <createDependencyReducedPom>true</createDependencyReducedPom>
+                    <filters>
+                        <filter>
+                            <artifact>*:*</artifact>
+                            <excludes>
+                                <exclude>META-INF/*.SF</exclude>
+                                <exclude>META-INF/*.DSA</exclude>
+                                <exclude>META-INF/*.RSA</exclude>
+                            </excludes>
+                        </filter>
+                    </filters>
+                </configuration>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <transformers>
+                                <transformer
+                                    implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer" />
+                                <transformer
+                                    implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                                    <mainClass>com.spotify.reaper.ReaperApplication</mainClass>
+                                </transformer>
+                            </transformers>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>com.spotify</groupId>
+                <artifactId>docker-maven-plugin</artifactId>
+                <version>1.0.0</version>
+                <configuration>
+                    <imageName>cassandra-reaper</imageName>
+                    <dockerDirectory>src/main/docker</dockerDirectory>
```

Comment:
Running `mvn clean prackage docker:build` from the top level fails to build the docker image. I think this is because this path here needs to be _server/src/main/docker_. When this path is updated to that value, the "Cassandra Reaper server" artifact builds successfully. However, the "Cassandra Reaper project" fails to build.
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] Cassandra Reaper server ............................ SUCCESS [  8.901 s]
[INFO] Cassandra Reaper project ........................... FAILURE [  1.900 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.042 s
[INFO] Finished at: 2017-08-25T15:41:32+10:00
[INFO] Final Memory: 24M/370M
[INFO] ------------------------------------------------------------------------
[ERROR] No plugin found for prefix 'docker' in the current project and in the plugin groups [org.apache.maven.plugins, org.codehaus.mojo] available from the repositories [local (/Users/anthony/.m2/repository), central (https://repo.maven.apache.org/maven2)] -> [Help 1]
```

In addition, changing the `<dockerDIrectory>` value to _server/src/main/docker_ means that `mvn clean prackage docker:build` when called within the _server_ directory. Unsure if that is an issue though.

Reply:
> Running mvn clean package docker:build from the top level fails to build the docker image.

I'm changing my mind on this.
Running `mvn docker:build` is only applicable to server/pom.xml and so should be done in server/